{"name": "test_books_with_videos_solutions_tile", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nNoSuchElementError: An element could not be located on the page using the given search parameters.\n    at AndroidUiautomator2Driver.findElOrEls (/Users/lekhraj/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.ts:86:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at AndroidUiautomator2Driver.findElOrElsWithProcessing (/Users/lekhraj/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)\n    at AndroidUiautomator2Driver.findElement (/Users/lekhraj/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12)", "trace": "self = <Testcases.test_E2E.TestEmbibeApp object at 0x104260d30>\n\n    @pytest.mark.usefixtures(\"appiumdriver\", \"log_on_failure\")\n    def test_books_with_videos_solutions_tile(self):\n        self._login_with_password()\n        LH = LearnHome(self.driver)\n>       LH.books_with_videos_solutions()\n\nTestcases/test_E2E.py:90: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nPages/learnhome.py:123: in books_with_videos_solutions\n    self.driver.find_element(AppiumBy.XPATH, '(//android.widget.FrameLayout[@resource-id=\"com.embibe.student:id/adBannerCardView\"])[1]').click()\n../../Library/Python/3.10/lib/python/site-packages/appium/webdriver/webdriver.py:382: in find_element\n    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']\n../../Library/Python/3.10/lib/python/site-packages/selenium/webdriver/remote/webdriver.py:380: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <appium.webdriver.errorhandler.MobileErrorHandler object at 0x104265030>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"An element could not be located on the page u...tomator2Driver.findElement (/Users/lekhraj/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12)\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"\n        https://www.w3.org/TR/webdriver/#errors\n        \"\"\"\n        payload = response.get('value', '')\n        if isinstance(payload, dict):\n            payload_dict = payload\n        else:\n            try:\n                payload_dict = json.loads(payload)\n            except (json.JSONDecodeError, TypeError):\n                return\n            if not isinstance(payload_dict, dict):\n                return\n        value = payload_dict.get('value')\n        if not isinstance(value, dict):\n            return\n        error = value.get('error')\n        if not error:\n            return\n    \n        message = value.get('message', error)\n        stacktrace = value.get('stacktrace', '')\n        # In theory, we should also be checking HTTP status codes.\n        # Java client, for example, prints a warning if the actual `error`\n        # value does not match to the response's HTTP status code.\n        exception_class: Type[sel_exceptions.WebDriverException] = ERROR_TO_EXC_MAPPING.get(\n            error, sel_exceptions.WebDriverException\n        )\n        if exception_class is sel_exceptions.WebDriverException and message:\n            if message == 'No such context found.':\n                exception_class = appium_exceptions.NoSuchContextException\n            elif message == 'That command could not be executed in the current context.':\n                exception_class = appium_exceptions.InvalidSwitchToTargetException\n    \n        if exception_class is sel_exceptions.UnexpectedAlertPresentException:\n            raise sel_exceptions.UnexpectedAlertPresentException(\n                msg=message,\n                stacktrace=format_stacktrace(stacktrace),\n                alert_text=value.get('data'),\n            )\n>       raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))\nE       selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       NoSuchElementError: An element could not be located on the page using the given search parameters.\nE           at AndroidUiautomator2Driver.findElOrEls (/Users/lekhraj/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.ts:86:11)\nE           at processTicksAndRejections (node:internal/process/task_queues:95:5)\nE           at AndroidUiautomator2Driver.findElOrElsWithProcessing (/Users/lekhraj/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)\nE           at AndroidUiautomator2Driver.findElement (/Users/lekhraj/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12)\n\n../../Library/Python/3.10/lib/python/site-packages/appium/webdriver/errorhandler.py:125: NoSuchElementException"}, "start": 1732465716492, "stop": 1732465783480, "uuid": "0d2ab9d0-75d6-420d-b4c6-aa8be1436280", "historyId": "ed03711f8c58355e17e854fee0e1d5b0", "testCaseId": "ed03711f8c58355e17e854fee0e1d5b0", "fullName": "Testcases.test_E2E.TestEmbibeApp#test_books_with_videos_solutions_tile", "labels": [{"name": "tag", "value": "@pytest.mark.usefixtures('appiumdriver', 'log_on_failure')"}, {"name": "parentSuite", "value": "Testcases"}, {"name": "suite", "value": "test_E2E"}, {"name": "subSuite", "value": "TestEmbibeApp"}, {"name": "host", "value": "EBL-C02H385BQ6L4.local"}, {"name": "thread", "value": "27800-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Testcases.test_E2E"}]}